gitlab/v4/objects/files.py

```diff
@@ -131,7 +127,37 @@ def get(  # type: ignore
         Returns:
             The generated RESTObject
         """
-        return cast(ProjectFile, GetMixin.get(self, file_path, ref=ref, **kwargs))
+        if TYPE_CHECKING:
+            assert file_path is not None
+        file_path = utils.EncodedId(file_path)
+        path = f"{self.path}/{file_path}"
+        server_data = self.gitlab.http_get(path, ref=ref, **kwargs)
+        if TYPE_CHECKING:
+            assert isinstance(server_data, dict)
+        return self._obj_cls(self, server_data)
+
+    def head(
+        self, file_path: str, ref: str, **kwargs: Any
+    ) -> "requests.structures.CaseInsensitiveDict[Any]":
+        """Retrieve just metadata for a single file.
+
+        Args:
+            file_path: Path of the file to retrieve
+            ref: Name of the branch, tag or commit
+            **kwargs: Extra options to send to the server (e.g. sudo)
+
+        Raises:
+            GitlabAuthenticationError: If authentication is not correct
+            GitlabGetError: If the file could not be retrieved
+
+        Returns:
+            The generated RESTObject
```

Comment:
Maybe something like this would be more accurate, and then we're good to go:

```suggestion
            The response headers as a dictionary
```